src/lib.rs

```diff
@@ -21,6 +21,11 @@ pub mod errors {
         pub code: ::reqwest::StatusCode
     }
 
+    #[derive(Fail, Debug)]
+    #[fail(display = "the following downloads failed:\n{}", _0)]
+    /// Files which are failed to download
+    pub struct FilesDownloadFailed(String);
```

Comment:
`String` as a parameter is too broad of a cause. Semantically, this error should be reporting which files are causing the download to fail--maybe a list of ids or something.

Reply:
> `String` as a parameter is too broad of a cause. Semantically, this error should be reporting which files are causing the download to fail--maybe a list of ids or something.

Agree, so I use a list of `(PathBuf, String)` which means files and the corresponding error message.